### ISSUE TITLE
Allow us to disable non-important things like bandwidth monitoring from GUI

### ISSUE THREAD

**Issue by ghost** [ericpaulbishop/gargoyle#697]
Personally, I have no need to monitor bandwidth because I have no data cap and don't wish to monitor anybody on my network.  More importantly, I don't need the added resource usage. My router isn't getting an upgrade anytime soon. 

The ability to disable bandwidth monitoring easily from the GUI as well as any other non-important thing that runs on startup would make for a much more customizable, fast experience using Gargoyle.

**Comment by rpodric** [comment]
Sure, but wouldn't LEDE (soon to merge back into OpenWRT, last I read) be a better fit then?  Gargoyle's raison d'etre is bandwidth monitoring. Why swim upstream, so to speak?

**Comment by Cormak** [comment]
@rpodric LEDE is fairly advanced for most people. Gargoyle is much more user-friendly and much more likely to be installed than LEDE. Same reason why you see most people installing DDWRT over OpenWRT, for example. 

Furthermore, another major benefit of Gargoyle is the stability it offers. How many times do you see people try to install a user-friendly option like DDWRT only to experience bugs when they could have installed Gargoyle and had a relatively bug-free experience? Many times!

I'm not saying that Gargoyle should go backwards. I'm just saying that Gargoyle should have a user-friendly option to disable things that some users don't really need, and in some cases disabling bandwidth monitoring can make a big difference in performance while still allowing users  enjoy the other many benefits and ease of using Gargoyle. 

See the plugins page on Gargoyle's UI? There should be a disable/uninstall button on bandwidth monitoring as well so that people can customize their install and get a little more out of their installation along with it. 

Other things should be easily disabled as well. Gargoyle should be as simple to customize by removing stuff as well as adding plugins to it so everybody can be happy.

**Comment by d3fz** [comment]
That's already possible with the "Services" plugin and a minor tweak to the initd.js file. By default, most of Gargoyle system features/services are hidden/blocked, so the average user can't shoot himself in the foot if he doesn't know what he's doing. The thing is, if the user **knows** what he's doing, and want to customize Gargoyle a little further (everyone has different needs), the only option is through the command line, which isn't a bad thing, just not as practical/easy as a few clicks directly through the UI. 

If you feel like giving it a try, install the "Services" plugin, SSH into the router and modify this file `/plugin_root/www/js/initd.js`, by editing `var blockedServices` under `resetData()`.

Services hidden/blocked by default:
`var blockedServices = ['boot','bwmon_gargoyle','cron','ddns_gargoyle','dnsmasq','done','dropbear','firewall','fstab','gargoyle_themes','led','log','network','qos_gargoyle','relayd','set_kernel_timezone','share_users','sysctl','sysfixtime','sysntpd','system','telnet','uhttpd','umount','usb_storage','usbmode','webmon_gargoyle','wol'];
`

Whether or not which services can/should be "unblocked", it's another topic.

**Comment by Cormak** [comment]
> Whether or not which services can/should be "unblocked"

Anything that runs on startup and isn't essential to the router's daily uptime. 

Simple.

**Comment by tacuar** [comment]
@Cormak It seems the latest version has removed the option to disable the bandwidth monitoring feature. I still have one router running the previous old Gargoyle-1.8.1 version and actually the default setting of bandwidth monitoring is disabled. If i need this running i just can make it enabled. I think this is good. However the old version is based on openwrt 1407 barrier breaker and thus not so up to latest security patches.

But if you mainly pursue for speed, you might try that 1.8.1

**Comment by lantis1008** [comment]
Are you certain?
My recollection is that it was ALWAYS enabled.

**Comment by d3fz** [comment]
Pretty sure it's always been enabled, otherwise the Quotas feature wouldn't work out-of-the-box. Correct?

**Comment by Cormak** [comment]
@tacuar It's definitely always been enabled and I would never risk my network by downgrading to a much older firmware for the sake of speed. This should be a user-friendly option so that people can have Gargoyle do as much or as little as they want.

**Comment by d3fz** [comment]
Like this?

![914b52b2-fee7-4880-90b2-56c57d5b8b3c](https://user-images.githubusercontent.com/22578839/34178936-1801bf18-e4f0-11e7-9bef-2166737b0136.png)

**Comment by lantis1008** [comment]
I am reluctant to allow such functionality. 
All it will result in is people shooting themselves in the foot. 

NACK from me.

**Comment by Cormak** [comment]
@d3fz Exactly like that as long as unchecking autostart actually means it is permanently disabled no matter how many times I restart the router. 

@lantis1008 You can only shoot yourself in the foot if you go looking for it. Bandwidth monitoring is not essential to the router so even if somebody does it by accident, who cares? This isn't like deleting system32 on Windows XP. You can simply enable it again.

**Comment by lantis1008** [comment]
Yes, but it is in the mission statement for Gargoyle (the first one even). 

I’ve given my opinion, I don’t agree with it, so I’m not going to contribute to making it a reality. 
If someone else wants to though, I’m happy to help make sure it is done right. 
It’s not me that gives the final yes or no on anything anyway. :)

**Comment by fengchen-github** [comment]
I  think it is necessary to add a switch button to "enable / disable" Bandwidth Usage, Bandwidth Distribution and QoS Distribution too.The most useful thing of Gargoyle for me is QoS.
